designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,209 @@
+
+Rationale:
+* users have declared their intention to deploy an application, at this point we do not want to impede them
+* logged metrics will help inform how often the Toolkit fails to determine an association
+* messaging the user could be overly spammy if they cannot affect their environment (example: an account with Lambda permissions but not ResourceGroupsTaggingAPI permissions)
+
+### Unit Tests
+
+* Deleting a Lambda Function
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a Lambda Function that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to delete a Lambda Function that is associated with a pipeline
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a Lambda Function and pipeline associations cannot be determined
+* Deleting a CloudFormation Stack
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a CloudFormation Stack that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to delete a CloudFormation Stack that is associated with a pipeline
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a CloudFormation Stack and pipeline associations cannot be determined
+* Deploying a SAM Application
+  * it does not prompt the user with the pipeline-related confirmation when attempting to deploy a SAM Application to a CloudFormation Stack that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to deploy a SAM Application to a CloudFormation Stack that is associated with a pipeline
+  * it continues to prompt the user if they attempt to select the same stack again
+  * it does not prompt the user with the pipeline-related confirmation when attempting to deploy a SAM Application to a CloudFormation Stack and pipeline associations cannot be determined
+
+### System Tests
+
+The Toolkit does not have System Tests at this time. System Tests task(s) will be added to the backlog, to be worked on after a Systems Level testing has been designed and created for the Toolkit.
+
+* it prompts the user for confirmation when attempting to delete a Lambda Function that is associated with a pipeline
+* it prompts the user for confirmation when attempting to delete a CloudFormation Stack that is associated with a pipeline
+* it prompts the user for confirmation when attempting to deploy a SAM Application to an existing CloudFormation Stack that is associated with a pipeline
+
+### Other Design Decisions
+
+* Caching information on the Explorer (like pipeline associations) exposes the Toolkit to edge case scenarios (timing issues, data volume, state freshness). The Toolkit will query on an as-needed basis at this time.
```

Comment:
Our explorer data only currently updates if you refresh or directly delete a resource (for instance, deleting a CloudFormation stack will immediately wipe it from the CloudFormation section but only will remove it from the Lambda section on refresh). I think that this cadence should be fine--we should never rely on the cached state.

Reply:
+1